cli/internal/core/engine.go

```diff
@@ -267,6 +248,12 @@ func (e *Engine) AddDep(fromTaskID string, toTaskID string) error {
 	if fromPkg != ROOT_NODE_NAME && fromPkg != util.RootPkgName && !e.TopologicGraph.HasVertex(fromPkg) {
 		return fmt.Errorf("found reference to unknown package: %v in task %v", fromPkg, fromTaskID)
 	}
-	e.PackageTaskDeps = append(e.PackageTaskDeps, []string{fromTaskID, toTaskID})
+
+	if _, ok := e.PackageTaskDeps[fromTaskID]; !ok {
+		e.PackageTaskDeps[toTaskID] = []string{}
```

Comment:
I think we need to make sure the map isn't nil. Appending to a nil slice is valid, but assigning an element to a nil map isn't.

Reply:
How would `e.PackageTaskDeps` would ever be nil? 

I had the same question about why we need this:

https://github.com/vercel/turbo/blob/652408b2167453c5f325ef4d0b7ffd351a386c8a/cli/internal/core/engine.go#L112-L114

but doesn't the compiler guarantee that it's not nil? (changed to map in this PR):
https://github.com/vercel/turbo/blob/652408b2167453c5f325ef4d0b7ffd351a386c8a/cli/internal/core/engine.go#L32

Reply:
`PackageTaskDeps [][]string` is specifying the type, that's not an initialization. It's instead a typed nil.